src/openassetio-core/specification/Specification.cpp

```diff
@@ -51,6 +60,8 @@ Specification::Specification(const TraitIds& traitIds) : impl_{std::make_unique<
 
 Specification::~Specification() = default;
 
+Specification::TraitIds Specification::traitIds() const { return impl_->traitIds(); }
```

Comment:
[question] Out of date `cpp` brain, is the compiler clever here and "do the right things and not copy the vector umpteen times"?

Reply:
Yeah, there will likely be one copy then one move.  

In one of my earlier sketches I imagined a specification's trait set as being statically defined, so you could safely return `const TraitIds&` and avoid any copies or moves.  But that is a bit more restrictive.

I'm happy with your more flexible solution and we can tweak later if performance is a problem.

Reply:
Cool, thanks. I think we definitely can't assume they will be static, eg [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/301/files#diff-29adf34bb4665be39936d3cbfcf6d37b4659d84d8fcb4a218f49fe75151079ffR20), and "the photoshop example".